tests/integration/release/parallel_test_launcher.py

```diff
@@ -0,0 +1,218 @@
+import csv
+import glob
+import os
+import shutil
+import subprocess
+import time
+from collections.abc import Sequence
+from datetime import datetime
+from pathlib import Path
+
+import pandas as pd
+
+
+def create_slurm_script(row: dict[str, str], script_name: str, output_dir: str) -> None:
+    dataset = row["dataset"]
+    engine = row["engine"]
+    memory = row["memory"]
+    time_limit = row["time"]  # DD:HH:MM
+    partition = (
+        "all.q" if int(time_limit.split(":")[0]) <= 24 else "long.q"
+    )  # long.q if 24 longer than 24 hours
+    release_tests_dir = Path(__file__).parent
+    pytest_command = (
+        f"pytest -rA --release --dataset {dataset} --engine {engine} {release_tests_dir}"
+    )
+    # TODO: define cpus per task based on engine (ask Zeb)
+    slurm_script = f"""#!/bin/bash 
+#SBATCH --job-name=pytest_{dataset}_{engine} 
+#SBATCH --output={output_dir}/pytest_{dataset}_{engine}.out 
+#SBATCH --error={output_dir}/pytest_{dataset}_{engine}.err 
+#SBATCH --mem={memory}G
+#SBATCH --time={time_limit} 
+#SBATCH --account=proj_simscience_prod
+#SBATCH --partition={partition}
+#SBATCH --cpus-per-task=1
+        
+echo "Running pytest for dataset {dataset} with engine {engine}" 
+{pytest_command} 
+    """
+    # Ensure the directory we want to write to exists
+    script_path = f"{output_dir}/tmp_scripts/{script_name}"
+    directory = os.path.dirname(script_path)
+    os.makedirs(directory, exist_ok=True)
+
+    # Write the script to a file
+    with open(script_path, "w") as script:
+        script.write(slurm_script)
+
+
+def submit_slurm_job(script_name: str) -> str | None:
+    try:
+        result = subprocess.run(
+            f"sbatch {script_name}", shell=True, check=True, capture_output=True
+        )
+        print(f"Submitted job: {script_name}")
+
+        # Extract the job ID from the output
+        output = result.stdout.strip()
+        print(f"Submission successful: {output.decode('utf-8')}")
+
+        # Slurm output is typically in the format "Submitted batch job <job_id>"
+        job_id = output.split()[-1]  # Extract the last word (the job ID)
+        return job_id.decode("utf-8")  # return bytes as string
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to submit job: {script_name}\nError: {e}")
+        return None
+
+
+def is_job_running(job_id: str) -> bool:
+    """
+    Checks if a job with the given Slurm job ID is still running or pending.
+
+    Args:
+        job_id (str): Slurm job ID.
+
+    Returns:
+        bool: True if the job is running or pending; False if it has finished or if checking the job status fails.
+    """
+    try:
+        result = subprocess.run(
+            f"squeue --job {job_id}", shell=True, capture_output=True, text=True
+        )
+        return (
+            job_id in result.stdout
+        )  # The job ID is listed in squeue if it's running/pending
```

Comment:
It'll also show up there in a few other states, e.g. suspended, canceled (but not yet removed), etc. Not sure if that functionally matters or not though.